x/spec/types/spec.go

```diff
@@ -7,6 +7,97 @@ import (
 
 const minCU = 1
 
+type getterFunc func(index string) (Spec, bool)
+
+// ExpandSpec expands the "imports" field of the spec -if needed, recursively-
+// and adds to the current Spec those additional APIs from the imported Spec(s).
+// It takes as argument a getter function to fetch imported Spec(s) by name from
+// the keeper store, on demand. (Can't import the keeper module due to circular
+// dependencies)
+func (spec *Spec) ExpandSpec(getSpec getterFunc) (map[string]string, error) {
+	depends := map[string]bool{spec.Index: true}
+	details, err := spec.doExpandSpec(getSpec, depends, spec.Index)
+	return map[string]string{"spec": spec.Name, "imports": details}, err
+}
+
+// doExpandSpec performs the actual work and recusion for ExpandSpec above.
+func (spec *Spec) doExpandSpec(getSpec getterFunc, depends map[string]bool, details string) (string, error) {
+	if len(spec.Imports) == 0 {
+		return details, nil
+	}
+
+	var parents []Spec
+
+	// visual markers when import deepens
+	details += "->["
+
+	// recursion to get all parent specs
+	comma := ""
+	for _, index := range spec.Imports {
```

Comment:
should be extracted in a method for readability

Reply:
Can you detail you have in mind?  Moving the inside of this loop to a helper function, means that the recursion will ping-pong between this function and the helper, and I think that's less readable.

Reply:
i would move it to a function that performs fetch parent apiInterfaces, its cleaner and clearer, there is no reason to recurse the combination of the apis to one list so you can only recurse that one

Reply:
The recursion is standard DFS and I prefer to not split into two functions that call each other.  Will try to move the spec fetch and validation to a helper function.